### PR TITLE
Feeder Rework

### PR DESCRIPTION
**Pull request by AntarcticaByToto**
Changed feeder to use one banner sensor and implemented current draw to detect second ball using deque

### CODE REVIEW

File: Competition/src/main/cpp/subsystems/Feeder.cpp [hunk 1]
```diff
@@ -32,19 +30,20 @@ void Feeder::init()
...
-    table->PutNumber("Feeder Reverse Speed", FeederConstants::DEFUALT_FEEDER_SPEED_REVERSE);
-    table->PutNumber("Intake Forward Speed", FeederConstants::DEFUALT_INTAKE_SPEED_FORWARD);
-    table->PutNumber("Feeder Forward Speed", FeederConstants::DEFUALT_FEEDER_SPEED_FORWARD);
+    table->PutNumber("Intake Reverse Speed", FeederConstants::DEFAULT_INTAKE_SPEED_REVERSE);
+    table->PutNumber("Feeder Reverse Speed", FeederConstants::DEFAULT_FEEDER_SPEED_REVERSE);
+    table->PutNumber("Intake Forward Speed", FeederConstants::DEFAULT_INTAKE_SPEED_FORWARD);
+    table->PutNumber("Feeder Forward Speed Default", FeederConstants::DEFAULT_FEEDER_SPEED_FORWARD_DEFAULT);
+    table->PutNumber("Feeder Forward Speed Shoot", FeederConstants::DEFAULT_FEEDER_SPEED_FORWARD_SHOOT);
+    
+    state.spiked = false;
```
Comment (mray190) [inline Competition/src/main/cpp/subsystems/Feeder.cpp:44]: Call resetState here instead to make sure logic is consistent

File: Competition/src/main/cpp/subsystems/Feeder.cpp [hunk 2]
```diff
@@ -72,70 +71,117 @@ void Feeder::assessInputs()
...
+        state.current_cache.push_back(0);
+    }
+}
+
+
 void Feeder::resetState()
 {
     state.feederState = FeederState::FEEDER_DISABLE;
+
+    state.spiked = false;
```
Comment (mray190) [inline Competition/src/main/cpp/subsystems/Feeder.cpp:184]: Add previousBanner = false

File: Competition/src/main/cpp/subsystems/Feeder.cpp [hunk 3]
```diff
@@ -72,70 +71,117 @@ void Feeder::assessInputs()
...
 
-    state.lowerBannerTripped = false;//banner_upper.Get();
-    state.upperBannerTripped = false;//banner_upper.Get();
+    state.bannerTripped = !banner.Get();
 
-    if (state.driver_rightBumperPressed || state.operator_leftBumperPressed) {
-        state.feederState = FeederState::FEEDER_INTAKE2;
+    state.currentBanner = state.bannerTripped;
+    
+    if (state.operator_leftBumperPressed) {
```
Comment (mray190) [inline Competition/src/main/cpp/subsystems/Feeder.cpp:78]: Not code related, but the driver/operator diagrams need to be updated

File: Competition/src/main/cpp/subsystems/Feeder.cpp [hunk 4]
```diff
@@ -72,70 +71,117 @@ void Feeder::assessInputs()
...
-        //     state.feederState = state.upperBannerTripped ? FeederState::FEEDER_DISABLE : FeederState::FEEDER_INTAKE2;
-        // } 
-        // else if (state.upperBannerTripped) {
-        //     state.feederState = FeederState::FEEDER_INTAKE1;
-        // }
-        // else {
-            state.feederState = FeederState::FEEDER_INTAKE1;
-       // }
+    else if (state.operator_aButtonPressed || state.driver_rightBumperPressed) {
+        if (state.bannerTripped) {
```
Comment (mray190) [inline Competition/src/main/cpp/subsystems/Feeder.cpp:87]: Logic needs to be moved to assignOutputs or analyzeDashboard in order for it to work in Auto

**Review by mray190 — CHANGES_REQUESTED** [review]
Looks good except for a few items that will impact Auto. See comments above

File: Competition/src/main/cpp/subsystems/Feeder.cpp (hunk 4 above)
Comment (droiddoes9) [reply Competition/src/main/cpp/subsystems/Feeder.cpp:87]: made new auto_intake state and moved it to assign outputs

File: Competition/src/main/cpp/subsystems/Feeder.cpp (hunk 3 above)
Comment (droiddoes9) [reply Competition/src/main/cpp/subsystems/Feeder.cpp:78]: done

File: Competition/src/main/cpp/subsystems/Feeder.cpp (hunk 2 above)
Comment (droiddoes9) [reply Competition/src/main/cpp/subsystems/Feeder.cpp:184]: done

File: Competition/src/main/cpp/subsystems/Feeder.cpp (hunk 1 above)
Comment (droiddoes9) [reply Competition/src/main/cpp/subsystems/Feeder.cpp:44]: replaced the 3 manual resets with 1 call to reset state

**Pull request merged**